pcdsdevices/interface.py

```diff
@@ -93,6 +95,233 @@ def __repr__(self):
+            the only other key in the dictionary will be value.
+        """
+        def subdevice_filter(info):
+            return bool(info['kind'] & Kind.normal)
+
+        return ophydobj_info(self, subdevice_filter=subdevice_filter)
+
+
+def get_name(obj, default):
+    try:
+        return obj.name
+    except AttributeError:
+        try:
+            return str(obj)
+        except Exception:
+            return default
+
+
+def get_kind(obj):
+    try:
+        return obj.kind
+    except Exception:
+        return Kind.omitted
+
+
+def get_value(signal):
+    try:
+        # Minimize waiting, we aren't collecting data we're showing info
+        if signal.connected:
+            return signal.get(timeout=0.1, connection_timeout=0.1)
+    except Exception:
+        pass
+    return None
+
+
+def ophydobj_info(obj, subdevice_filter=None, devices=None):
+    if isinstance(obj, Signal):
+        return signal_info(obj)
+    elif isinstance(obj, Device):
+        return device_info(obj, subdevice_filter=subdevice_filter,
+                           devices=devices)
+    else:
+        return {}
+
+
+def device_info(device, subdevice_filter=None, devices=None):
+    if devices is None:
+        devices = set()
+    name = get_name(device, default='device')
+    kind = get_kind(device)
+    info = dict(name=name, kind=kind, is_device=True)
+
+    try:
+        # Show the current preset state if we have one
+        # This should be the first key in the ordered dict
+        has_presets = device.presets.has_presets
+    except AttributeError:
+        has_presets = False
+    if has_presets:
+        try:
+            info['preset'] = device.presets.state()
+        except Exception:
+            info['preset'] = 'ERROR'
+
+    try:
+        # Extra key for positioners
+        # This has ordered dict priority over everything but the preset state
+        info['position'] = device.position
+    except AttributeError:
+        pass
+    except Exception:
+        # Something else went wrong! We have a position but it didn't work
+        info['position'] = 'ERROR'
+
+    if device not in devices:
+        devices.add(device)
+        for cpt_name, cpt_desc in device._sig_attrs.items():
+            # Skip lazy signals outright in all cases
+            # Usually these are lazy because they take too long to getattr
+            if cpt_desc.lazy:
+                continue
+            # Skip attribute signals
+            # Indeterminate get times, no real connected bool, etc.
+            if issubclass(cpt_desc.cls, AttributeSignal):
+                continue
+            # Skip not implemented signals
+            # They never have interesting information
+            if issubclass(cpt_desc.cls, NotImplementedSignal):
+                continue
+            try:
+                cpt = getattr(device, cpt_name)
+            except AttributeError:
+                # Why are we ever in this block?
```

Comment:
Sure, I can put a message in here. Maybe a debug statement? I want to avoid it being noisy during usage since it will interfere with the clean printout.